src/main/java/com/google/sps/servlets/UserDataServlet.java

```diff
@@ -118,12 +152,39 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         userEntity.setProperty(USER_FRIENDS_LIST_PROPERTY, Arrays.asList(friends));
       }
     }
+    getAndStoreBlobKeys(request, userEntity);
     datastore.put(userEntity);
 
     // Redirect to the profile page, and let the front-end know the current logged in user
     response.sendRedirect("/profile.html?id=" + userId);
   }
 
+  /** Method that stores the blob-keys (in Datastore) of files uploaded to Blobstore */
+  private void getAndStoreBlobKeys(HttpServletRequest request, Entity userEntity) {
+    // Get values determining whether or not to update/save blobkeys of any particular image
+    String profilePhotoUploaded = getStringParameter(request, USER_PROFILE_PHOTO, IMAGE_NOT_FOUND);
+    String photo2Uploaded = getStringParameter(request, USER_PHOTO_2, IMAGE_NOT_FOUND);
+    String photo3Uploaded = getStringParameter(request, USER_PHOTO_3, IMAGE_NOT_FOUND);
+    String photo4Uploaded = getStringParameter(request, USER_PHOTO_4, IMAGE_NOT_FOUND);
+    String photo5Uploaded = getStringParameter(request, USER_PHOTO_5, IMAGE_NOT_FOUND);
```

Comment:
It seems like here that these will only ever have the string values of "true" or "false". Given that, is there a way to do this using boolean values instead of strings?